### PR TITLE
Require login

### PR DESCRIPTION
**Pull request by trainerbill**
New PR for requiring client side roles.  Props to @igorauad for the idea.

### CODE REVIEW

**Comment by rhutchison** [comment]
Supersedes https://github.com/meanjs/mean/pull/686

Great work @igorauad and @trainerbill 

LGTM @lirantal

File: modules/core/client/app/init.js [hunk 1]
```diff
@@ -10,6 +10,30 @@ angular.module(ApplicationConfiguration.applicationModuleName).config(['$locatio
...
+            toState.data.roles.forEach(function (role) {
+               if (Authentication.user.roles !== undefined && Authentication.user.roles.indexOf(role) !== -1) {
+                   allowed = true;
+                   return true;
+               }
+            });
+
+            if (!allowed) {
+                event.preventDefault();
+                $state.go('authentication.signin', {}, {
```
Comment (mleanos) [inline modules/core/client/app/init.js]: Is there a reason why this sends the user to the signin route? Shouldn't it take them 'home'?

**Comment by lirantal** [comment]
Closed #686 
Thank you guys for the great work and collaboration on this.

**Comment by lirantal** [comment]
@mleanos looks like I missed your comment there before merging in. @trainerbill ?

**Pull request merged**